server/interfaces/interfaces.go

```diff
@@ -238,6 +239,74 @@ type WorkflowService interface {
 	ServeHTTP(w http.ResponseWriter, r *http.Request)
 }
 
+type GitProviders []GitProvider
+
+type GitProvider interface {
+	// MatchRepoURL returns whether a given repo URL should be handled by this
+	// provider.
```

Comment:
ubernit: don't think it matters much but to be thorough you'd want to comment about what happens if multiple match (probably first is chosen or something)

Reply:
Done